Pulse/AVCaptureDevice+Extensions.swift

```diff
@@ -84,17 +45,65 @@ extension AVCaptureDevice {
         }
     }
     
-    func toggleTorch(on: Bool) {
+    func toggleTorch(on: Bool) throws {
         guard hasTorch, isTorchAvailable else {
-            print("Torch is not available")
-            return
+            throw TorchError.torchNotAvailable
         }
-        do {
             try lockForConfiguration()
             torchMode = on ? .on : .off
             unlockForConfiguration()
-        } catch {
-            print("Torch could not be used \(error)")
+    }
+}
+
+extension AVCaptureDevice {
+    private func availableFormatsFor(preferredFps: Float64) -> [AVCaptureDevice.Format] {
+        var availableFormats: [AVCaptureDevice.Format] = []
```

Comment:
`availableFormats` no longer changes. You could declare it as a constant here:

`let availableFormats`